dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/readerOMC.py

```diff
@@ -1226,6 +1230,44 @@ def assign_variables_indexes(self):
         self.nb_integer_vars = index_integer_double
         self.find_calculated_variables()
 
+    def detect_z_only_used_internally(self):
+        map_num_eq_vars_defined = self.get_map_num_eq_vars_defined()
+        dicr_variable_to_dics_equation_dependencies = {}
+
+        discr_vars = []
+        for var in self.list_vars:
+            name = var.get_name()
+            if name not in map_var_name_2_addresses : continue
+            test_param_address(name)
+            address = map_var_name_2_addresses[name]
+            if "discreteVars" in address or "integerDoubleVars" in address or "booleanVars" in address:
+                discr_vars.append(name)
+
+        for f in self.list_func_16dae_c:
+            f_num_omc = f.get_num_omc()
+            name_var_eval = None
+
+            # for Modelica reinit equations, the evaluated var scan does not always work
+            # a fallback is to look at the variable defined in this case
+            if f_num_omc in map_num_eq_vars_defined.keys():
+                if len(map_num_eq_vars_defined[f_num_omc]) > 1:
+                    error_exit("   Error: Found an equation (id: " + f_num_omc+") defining multiple variables. This is not supported in Dynawo.")
+                name_var_eval = map_num_eq_vars_defined[f_num_omc] [0]
+
+            if name_var_eval is not None and name_var_eval in discr_vars:
+                for discr_var in discr_vars:
+                    if discr_var == name_var_eval: continue
+                    if "/* " + discr_var + " " in str(f.get_body()):
+                        if not discr_var in dicr_variable_to_dics_equation_dependencies:
+                            dicr_variable_to_dics_equation_dependencies[discr_var] = []
+                        dicr_variable_to_dics_equation_dependencies[discr_var].append(f_num_omc)
+
+        for discr_var in discr_vars:
+            if discr_var not in dicr_variable_to_dics_equation_dependencies and not is_when_condition(discr_var):
```

Comment:
The previous loop on self.lest_func_16_dae.c enables to detect z that impact other z.
Does the is_when_condition enables to detect the z that impact g?

Reply:
No, this is just to make to be really sure that no whenCondition is considered as discrete (as they are not Z variables)